---
title: "DGE"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Understanding Dataset

- using a real RNA-Seq datasetfrom Kenny PJ et.al, Cell Rep 2014
  - RNA-Seq performed on HEK293F cells that were transfected with a MOV10 transgene, siRNA known to knock down MOV10 expression, or non-specific siRNA
    - resulted in 3 conditions
      - MOV10 over expression (oe)
      - MOV10 knock down (kd)
      - irrelevant kd
- using this data, evaluate transcriptional patterns associated with perturbation of MOV10 expression
- researchers are investigating interactions between various genes involved in Fragile X syndrome, a disease in which there is aberrant production of FMRP protein
  - can lead to mental retardation, premature ovarian failure, autism, Parkinson's disease, developmental delays, etc.
  - MOV10 is a putative RNA helicase that is associated with FMRP in the context of the microRNA pathway

<div style="text-align:center">
<img src="C:/USers/durbe/Documents/repositories/figures/StemAway/R_09Jun_Replicates.png"alt="Replicates">
</div>

<br>

<div style="text-align:center">
<img src="C:/USers/durbe/Documents/repositories/figures/StemAway/R_09Jun_Hypothesis.png"alt="Hypothesis">
</div>

### Our Questions

1. What patterns of expression can we identify with the loss or gain of MOV10?
2. Are there any genes shared between the two conditions?

# Metadata

Information about the data.
- encompasses any kind of information that relates to our data

For the data used today:

- RNA was extracted from treated HEK293F cells
- cDNA libraries for this dataset are stranded and were generated using the TruSeq Stranded mRNA library prep kit from Illumina
- sequencing was carried out on the Illumina HSeq-2500 for 100bp single end reads
- ~40 million reads per sample were generated

### 1. What types of metadata are used in your experimental design (any experiment)?

### 2. What other kinds of metadata might a sequencing project generate?

### 3. Why is this type of information important?

# From Sequence Data to Count Matrix

- most common use for transcriptome data is to search for differentially expressed genes
  - aids in understanding molecular basis of phenotypic variation

### Steps to Obtain Gene Counts

In order to perform differential expression (DE) analysis on RNA-Seq data

<div style="text-align:center">
<img src="C:/USers/durbe/Documents/repositories/figures/StemAway/R_09Jun_Counts.png"alt="Count Matrix Steps">
</div>

- there are several tools for each step
- standard workflow iwht an alignment and a counting step 
  - recently, an alignment-free counting workflow using Salmon and Kallisto is more common
    - these tools generate an abundance estimate for genes
    
# Differential Expression Analysis

Walkthrough for an end-to-end gene-level RNA-seq differential expression workflow.See DEanalysis folder in Training folder.

1. normalization of counts

- DESeq2 calculates size factor sfor each sample using the median of ratios method 

2. transformation of counts

- works best for data that generally have the same range of variance at different ranges of mean values
- for RNA-Seq raw counts, variance grows with the mean - large counts dominate PCA
  - simple strategy: take the logarithm of the normalized count values plus a small pseudocount (for 0 counts) - now lowest counts dominate PCA
  - DESeq2 strategy: take the regularized log transform
  
3. quality assessment and exploratory analysis

- PCA
- Hierarchical Clustering
  - samples that show low correlation (<.8) represent outliers
  
<div style="text-align:center">
<img src="C:/USers/durbe/Documents/repositories/figures/StemAway/R_09Jun_Workflow.png"alt="Count Matrix Steps">
</div>

4. DESeq2 - differential expression analysis

- log fold change shrinkage for visualization and ranking
- p-values and adjusted p-values

5. exploring and exporting results
  
- MA-plot
- volcano plots
